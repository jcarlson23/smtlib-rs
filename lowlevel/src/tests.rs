use crate::{ast::Script, lexicon::SmtlibParse, parse::Parser};

#[test]
fn escaped_string() {
    insta::assert_ron_snapshot!(String::parse(&mut Parser::new(
        r#""Hello ""world"" this is cool!""#
    )));
}

#[test]
fn bubble_sort() {
    insta::assert_ron_snapshot!(Script::parse(include_str!("../examples/bubble_sort.smt2")));
}
