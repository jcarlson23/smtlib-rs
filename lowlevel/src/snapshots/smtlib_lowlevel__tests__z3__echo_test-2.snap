---
source: lowlevel/src/tests.rs
expression: "d.exec(&Command::Echo(r#\"\"Hello, unmatched paren! :)\"\"#.to_string()))?"
---
SpecificSuccessResponse(GetProofResponse(GetProofResponse(SpecConstant(String("\"Hello, unmatched paren! :)\"")))))
